Invoker.hpp

```diff
@@ -1,25 +1,45 @@
+#ifndef INVOKER_HEADER
+#define INVOKER_HEADER
+
+#ifdef _MSC_VER
+#undef __cplusplus
+#define __cplusplus _MSVC_LANG
+#endif
+
+#if __cplusplus > 201402L
 #include <type_traits> 
-#include <concepts>
 #include <functional>
+#if __cplusplus > 201703L
+#include <concepts>
+#endif // __cplusplus > 201703L
 
 template <typename Invocable>
 struct Invoker{
-    using Invocable_T = std::conditional_t<std::is_function_v<Invocable>, 
-    std::add_pointer_t<Invocable>, Invocable>;
-    template <typename I> 
+    using Invocable_T = std::conditional_t<std::is_function_v<Invocable>, std::add_pointer_t<Invocable>, Invocable>;
+#if __cplusplus > 201703L
+    template <typename I>
     requires std::convertible_to<I, Invocable_T>
-    Invoker(I invocable): 
-    invocable_(std::move(invocable)) {} 
+#else // __cplusplus < 202002L
+    template <typename I, typename = typename std::enable_if<std::is_convertible<I, Invocable_T>::value>::type>
```

Comment:
please change this to:
template <typename I, std::enable_if_t<std::is_convertible_v<I, Invocable_T>>>